components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/rabbitmq/RabbitMQConnectionFactory.java

```diff
@@ -131,184 +93,160 @@ private void handleException(String msg) {
+    public Connection createConnection() throws RabbitMQException {
         Connection connection = null;
         try {
             connection = RabbitMQUtils.createConnection(connectionFactory, addresses);
             log.info("[" + name + "] Successfully connected to RabbitMQ Broker");
         } catch (IOException e) {
-            log.error("[" + name + "] Error creating connection to RabbitMQ Broker. Reattempting to connect.", e);
-            int retryC = 0;
-            while ((connection == null) && ((retryCount == -1) || (retryC < retryCount))) {
-                retryC++;
-                log.info("[" + name + "] Attempting to create connection to RabbitMQ Broker" +
-                        " in " + retryInterval + " ms");
-                try {
-                    Thread.sleep(retryInterval);
-                    connection = RabbitMQUtils.createConnection(connectionFactory, addresses);
-                    log.info("[" + name + "] Successfully connected to RabbitMQ Broker");
-                } catch (InterruptedException e1) {
-                    log.error("[" + name + "] Error while trying to reconnect to RabbitMQ Broker", e1);
-                } catch (IOException e2) {
-                    log.error("[" + name + "] Error while trying to reconnect to RabbitMQ Broker", e2);
-                }
-            }
+            log.error("[" + name + "] Error creating connection to RabbitMQ Broker. " +
+                    "Reattempting to connect.", e);
+            connection = retry(connection);
             if (connection == null) {
-                handleException("[" + name + "] Could not connect to RabbitMQ Broker. Error while creating connection", e);
+                throw new RabbitMQException("[" + name + "] Could not connect to RabbitMQ Broker. " +
+                        "Error while creating connection", e);
             }
         }
         return connection;
     }
 
-    public int getRetryInterval() {
-        return retryInterval;
-    }
-
-    public int getRetryCount() {
-        return retryCount;
+    /**
+     * Retry when could not connect to the broker
+     *
+     * @param connection the failure connection {@link Connection} object
+     * @return the {@link Connection} object after retry completion
+     */
+    private Connection retry(Connection connection) {
+        int retryC = 0;
+        while ((connection == null) && ((retryCount == -1) || (retryC < retryCount))) {
+            retryC++;
+            log.info("[" + name + "] Attempting to create connection to RabbitMQ Broker" +
+                    " in " + retryInterval + " ms");
+            try {
+                Thread.sleep(retryInterval);
+                connection = RabbitMQUtils.createConnection(connectionFactory, addresses);
+                log.info("[" + name + "] Successfully connected to RabbitMQ Broker");
+            } catch (InterruptedException e1) {
+                Thread.currentThread().interrupt();
```

Comment:
I suppose the interruptedException is only thrown from the thread sleep above? In that case, the exception will be directly causght here without actually making an attempt to establish a connection and there can be a slight mismatch between the retry count logged and the actual retrie. Should we move the catch block to avoid this? OR should we not worry about this?

Reply:
The `InterruptedException` thrown terminating the thread. In this case, the server must shutdown while reconnecting happens. In such a case, we are marking the thread as interrupted. Is there any other way this could happen? I got the point you mentioned. But when server shutting down triggers, the control is not in our hands.